UnityProject/Assets/Scripts/UI/CameraZoomHandler.cs

```diff
@@ -0,0 +1,47 @@
+﻿using System;
+using UnityEngine;
+
+public class CameraZoomHandler : MonoBehaviour
+{
+	private int zoomLevel = 1;
+	private int cachedZoomLevel = 1;
+
+	void Start()
+	{
+		// Setup initial zoom level by guessing.
+		SetZoomLevel(-1);
+	}
+	
+	// Update is called once per frame
+	void Update()
+	{
+		if(cachedZoomLevel != zoomLevel)
+		{
+			// Calculate ratio.
+			double ratio = Screen.height / (double) Screen.width;
```

Comment:
Hm, what's the point of placing it in Update() if it's only getting changed via SetZoomLevel() anyway?
Why don't you convert this to Refresh() method that would recalculate camera size based on updated window size and just use this.zoomLevel ?